### ISSUE TITLE
[jit] `__hash__` magic method is missing

### ISSUE THREAD

**Issue by driazati** [pytorch/pytorch#34851]
```python
class X(object):
    def __init__(self):
        pass
    def __hash__(self):
        return 2

@torch.jit.script
def fn():
    x = X()
    return hash(x)

print(fn())
```

should work but errors out

**Comment by eellison** [comment]
Is this something you encountered or a something a user brought up ? I don't know how useful it is to enumerate all python features if no user is ever encountering it.

**Comment by driazati** [comment]
Just something we're missing. It's another random hole in the language that'd be pretty easy to fix, just like a lot of other stuff we found during the language survey

**Comment by eellison** [comment]
Makes sense. It would still be good to delineate user issues / real issues in some way.

**Comment by janeyx99** [comment]
TorchScript has been deprecated in favor of torch.export as of 2.11. https://pytorch.org/blog/pytorch-2-11-release-blog/ Closing this issue as we are not further developing on it--please ping if you believe this issue was erroneously closed.
